packages/medusa/src/api/routes/store/collections/list-collections.js

```diff
@@ -1,37 +1,108 @@
+ *     default: 20
+ *   - in: query
+ *     name: offset
+ *     schema:
+ *       type: integer
+ *     description: The index to begin retrieving Collections from.
+ *     default: 0
+ *   - in: query
+ *     name: q
+ *     schema:
+ *       type: string
+ *     description: Query based on title and handle of Collections.
+ *   - in: query
+ *     name: order
+ *     schema:
+ *        type: string
+ *     description: Order result based on either title or created_at / updated_at. Prefix '-' to indicate DESC.
+ *     default: title
  * tags:
  *   - Collection
  * responses:
- *  "200":
- *    description: OK
- *    content:
- *      application/json:
- *        schema:
- *          properties:
- *            collection:
- *              $ref: "#/components/schemas/product_collection"
+ *   200:
+ *     description: OK
+ *     content:
+ *       application/json:
+ *         schema:
+ *           properties:
+ *             count:
+ *               description: The total number of Collections.
+ *               type: integer
+ *             offset:
+ *               description: The offset for pagination.
+ *               type: integer
+ *             limit:
+ *               description: The maxmimum number of Collections to return.
+ *               type: integer
+ *             products:
+ *               type: array
+ *               items:
+ *                 $ref: "#/components/schemas/product_collection"
  */
 export default async (req, res) => {
-  const selector = {}
+  const schema = Validator.object({
+    limit: Validator.number().default(20),
+    offset: Validator.number().default(0),
+    q: Validator.string(),
+    tags: Validator.array().items(Validator.string()),
+    order: Validator.string()
+      .valid(
+        "title",
+        "-title",
+        "handle",
+        "-handle",
+        "created_at",
+        "-created_at",
+        "updated_at",
+        "-updated_at"
+      )
+      .default("title"),
+  })
 
-  const limit = parseInt(req.query.limit) || 10
-  const offset = parseInt(req.query.offset) || 0
+  const { value, error } = schema.validate(req.query)
+  if (error) {
+    throw new MedusaError(MedusaError.Types.INVALID_DATA, error.details)
+  }
 
-  const productCollectionService = req.scope.resolve("productCollectionService")
+  const collectionService = req.scope.resolve("productCollectionService")
+
+  const selector = {}
+
+  if (value.q) {
+    selector.q = value.q
+  }
 
   const listConfig = {
-    skip: offset,
-    take: limit,
+    relations: [],
+    skip: value.offset,
+    take: value.limit,
+  }
+
+  // We use a prefix of '-' to indicate a descending order type.
+  const order = value.order
+  listConfig.order = {
+    [order.replace("-", "")]: order[0] === "-" ? "DESC" : "ASC",
   }
 
-  const [collections, count] = await productCollectionService.listAndCount(
-    selector,
-    listConfig
-  )
+  const collections = await collectionService.list(selector, listConfig)
```

Comment:
Should be `listAndCount` so we get the actual global count based on the query

Reply:
https://vuuzletvph.com/heads-up-vuco-bounty-programs-are-coming-your-way/